website/public/js/controllers/inventoryCtrl.js

```diff
@@ -234,6 +241,72 @@ habitrpg.controller("InventoryCtrl",
       return filteredArray;
     };
 
+    $scope.groupEquipmentBy = function(group, costume) {
+
+     var gear = {};
+
+     if (group === "stat") {
+      $scope.gear.forEach(function(item, index, array) {
+       var index = "";
+       if ( item.int > 0 ) {
+        index = "Intelligence";
+        if (!gear[index]) gear[index] = [];
+        gear[index].push(item);
+       }
+       if ( item.per > 0 ) {
+        index = "Perception";
+        if (!gear[index]) gear[index] = [];
+        gear[index].push(item);
+       }
+       if ( item.con > 0 ) {
+        index = "Constitution";
+        if (!gear[index]) gear[index] = [];
+        gear[index].push(item);
+       }
+       if ( item.str > 0 ) {
+        index = "Strength";
+        if (!gear[index]) gear[index] = [];
+        gear[index].push(item);
+       }
+       if ( index === "" ) {
+        var index = "Undefined";//Use undefined her to order none last
+        if (!gear[index]) gear[index] = [];
+        gear[index].push(item);
+       }
+      });
+     } else if (group === "klass") {
+      $scope.gear.forEach(function(item, index, array) {
+       var index = "";
+       if (item.klass === "special") {
+        if (item.specialClass) {
+         index = $scope.capitalizeFirstLetter(item.specialClass) + " " + $scope.capitalizeFirstLetter(item.klass) ;
+        } else {
+         index = "Legendary";
+         //if there are no stats on the item, then we have a Fanciful item
+         if ( item.str == 0 && item.int == 0 && item.per == 0 && item.con == 0 ){
+          index = "Fanciful";
+         }
+        }
+        if (!gear[index]) gear[index] = [];
+        gear[index].push(item);
+       } else {
+        index = $scope.capitalizeFirstLetter(item.klass)
+        if (!gear[index]) gear[index] = [];
+        gear[index].push(item);
+       }
+      });
+     } else {
+      gear = _.groupBy($scope.gear, group);
+     }
+
+     if (costume != undefined) {
```

Comment:
Is there a reason `costume == false`, `costume == ''` and `costume == null` are okay?